lib/rouge/lexers/hcl.rb

```diff
@@ -0,0 +1,162 @@
+# -*- coding: utf-8 -*- #
+
+module Rouge
+  module Lexers
+    class Hcl < RegexLexer
+      tag 'hcl'
+
+      title 'Hashicorp Configuration Language'
+      desc 'Hashicorp Configuration Language, used by Terraform and other Hashicorp tools'
+
+      state :multiline_comment do
+        rule %r([*]/), Comment::Multiline, :pop!
+        rule %r([^*/]+), Comment::Multiline
+        rule %r([*/]), Comment::Multiline
+      end
+
+      state :comments_and_whitespace do
+        rule /\s+/, Text
+        rule %r(//.*?$), Comment::Single
+        rule %r(#.*?$), Comment::Single
+        rule %r(/[*]), Comment::Multiline, :multiline_comment
+      end
+
+      state :primitives do
+        rule /[0-9][0-9]*\.[0-9]+([eE][0-9]+)?[fd]?([kKmMgG]b?)?/, Num::Float
+        rule /[0-9]+([kKmMgG]b?)?/, Num::Integer
+
+        rule /"/, Str::Double, :dq
+        rule /'/, Str::Single, :sq
+        rule /(<<-?)(\s*)(\'?)(\\?)(\w+)(\3)/ do |m|
+          groups Operator, Text, Str::Heredoc, Str::Heredoc, Name::Constant, Str::Heredoc
+          @heredocstr = Regexp.escape(m[5])
+          push :heredoc
+        end
+      end
```

Comment:
Not quite, I just took the ordering found in the JavaScript lexer: https://github.com/jneen/rouge/blob/master/lib/rouge/lexers/javascript.rb. It kind of made sense since this set of keywords may be overridden in derived lexers (Terraform redefines these)

Reply:
I'm OK with this.